### PR TITLE
[squid:S2095] Resources should be closed

### PR DESCRIPTION
**Pull request by AymanDF**
This pull request is focused on resolving occurrences of Sonar rule 
squid:S2095 - “Resources should be closed”. 

You can find more information about the issue here: 
https://dev.eclipse.org/sonar/rules/show/squid:S2095

Please let me know if you have any questions.
Ayman Abdelghany.

### CODE REVIEW

File: src/org/stringtemplate/v4/NumberRenderer.java [hunk 1]
```diff
@@ -46,6 +46,8 @@ public String toString(Object o, String formatString, Locale locale) {
         if ( formatString==null ) return o.toString();
         Formatter f = new Formatter(locale);
         f.format(formatString, o);
-        return f.toString();
+        String str = f.toString();
```
Comment (sharwell) [inline src/org/stringtemplate/v4/NumberRenderer.java]: :bulb: This code should be in a try/finally for exception-safe closing of the `Formatter` instance.

**Comment by AymanDF** [comment]
Done

File: src/org/stringtemplate/v4/NumberRenderer.java [hunk 2]
```diff
@@ -44,8 +44,16 @@
     public String toString(Object o, String formatString, Locale locale) {
         // o will be instanceof Number
         if ( formatString==null ) return o.toString();
-        Formatter f = new Formatter(locale);
-        f.format(formatString, o);
-        return f.toString();
+        Formatter f = null;
```
Comment (sharwell) [inline src/org/stringtemplate/v4/NumberRenderer.java]: :bulb: Slight simplification

```java
Formatter f = new Formatter(locale);
try {
  f.format(formatString, o);
  return f.toString();
} finally {
  f.close();
}
```

**Comment by sharwell** [comment]
:+1:

**Comment by parrt** [comment]
doing similarly manually https://github.com/antlr/stringtemplate4/commit/40bfd1723a4f7e5985e090d4866bf0a2d8d4e68f . closing.

**Pull request closed without merging**